src/main/java/auto_scheduler/StudentInfoRepositoryImpl.java

```diff
@@ -1,7 +1,62 @@
 package org.txstate.auto_scheduler;
 
+
+import java.text.SimpleDateFormat;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.File;
+
+/**
+ * Creates a new StudentInfo object
+ *  Searches in the json files in records using the Student Id
+ */
+
+
 public class StudentInfoRepositoryImpl implements StudentInfoRepository{
-	public StudentInfo getInfo(String studentId) {
-		throw new UnsupportedOperationException("Implement by reading from json");
-	}
+     /**
+     * Searches json files in records by studentID
+     * If found return StudentInfo object
+     * If not found returns null
+     * @param studentId
+     * @return StudentInfo
+     */
+    public StudentInfo getInfo(String studentId) {
+
+        JSONParser parser = new JSONParser();
+        File folder = new File("src/main/resources/data/records/");
```

Comment:
we need to use the `ResourcePathProvider` interface to get the path of records folder. 
see sample implementation from [StudentRecordRepositoryImpl](https://github.com/samaniehsan/auto-scheduler/blob/master/src/main/java/auto_scheduler/StudentRecordRepositoryImpl.java)


```java
public StudentRecordRepositoryImpl(ResourcePathProvider pathProvider) {
		this.pathProvider = pathProvider;
	}
	
	public Collection<StudentRecord> get(String studentId) throws FileNotFoundException,  IOException {
		return getGrades(getFilePath(studentId), studentId);
	}


	private String getFilePath(String studentId) {
		String recordsFolderRelativePath = this.pathProvider.getRecords();
		String gradeCSVPath = recordsFolderRelativePath + "/" + studentId + "_grade.csv";
		return gradeCSVPath;

}
```

Reply:
I will be in class tonight. So, you can show me how to rebase it. I also will implement the ResourcePathProvider interface it when I get a chance.